tests/index.test.ts

```diff
@@ -1,154 +1,176 @@
 
-    expect(result[0].score).equal(150);
-    expect(result[4].score).equal(1000);
+    expect(result[0].age).equal(ASSERTIONS.MIN_AGE);
+    expect(result[myArr.length - 1].age).equal(ASSERTIONS.MAX_AGE);
   });
 
   it("ASC with num", function () {
-    let result = myArr.sortByAttr(a => a.score, 0);
+    let result = myArr.sortByAttr(a => a.age, 0);
 
-    expect(result[0].score).equal(150);
-    expect(result[4].score).equal(1000);
+    expect(result[0].age).equal(ASSERTIONS.MIN_AGE);
+    expect(result[myArr.length - 1].age).equal(ASSERTIONS.MAX_AGE);
   });
 
   it("DESC with enum ", function () {
-    let result = myArr.sortByAttr(a => a.score, SortDirection.DESC);
+    let result = myArr.sortByAttr(a => a.age, SortDirection.DESC);
 
-    expect(result[0].score).equal(1000);
-    expect(result[4].score).equal(150);
+    expect(result[0].age).equal(ASSERTIONS.MAX_AGE);
+    expect(result[myArr.length - 1].age).equal(ASSERTIONS.MIN_AGE);
   });
 
   it("DESC with num", function () {
-    let result = myArr.sortByAttr(a => a.score, 1);
+    let result = myArr.sortByAttr(a => a.age, 1);
 
-    expect(result[0].score).equal(1000);
-    expect(result[4].score).equal(150);
+    expect(result[0].age).equal(ASSERTIONS.MAX_AGE);
+    expect(result[myArr.length - 1].age).equal(ASSERTIONS.MIN_AGE);
   });
 
   it("ASC by string with num", function () {
-    let result = myArr.sortByAttr(a => a.nickName);
+    let result = myArr.sortByAttr(a => a.name.first);
 
-    expect(result[0].score).equal(1000);
-    expect(result[4].score).equal(999);
+    expect(result[0].name.first).equal(ASSERTIONS.LOWEST_NAME);
+    expect(result[myArr.length - 1].name.first).equal(ASSERTIONS.HIGHEST_NAME);
   });
 
   it("DESC by string with num", function () {
-    let result = myArr.sortByAttr(a => a.nickName, 1);
+    let result = myArr.sortByAttr(a => a.name.first, 1);
 
-    expect(result[0].score).equal(999);
-    expect(result[4].score).equal(1000);
+    expect(result[0].name.first).equal(ASSERTIONS.HIGHEST_NAME);
+    expect(result[myArr.length - 1].name.first).equal(ASSERTIONS.LOWEST_NAME);
   });
 
   it("Shuffle Then sort", function () {
     let result = myArr
       .shuffle()
       .shuffle()
-      .sortByAttr(a => a.score);
+      .sortByAttr(a => a.name.first);
 
-    expect(result[0].score).equal(150);
-    expect(result[4].score).equal(1000);
+    expect(result[0].name.first).equal(ASSERTIONS.LOWEST_NAME);
+    expect(result[myArr.length - 1].name.first).equal(ASSERTIONS.HIGHEST_NAME);
   });
 });
 
 describe("shuffle", () => {
-  it("Shuffle Then sort", function () {
+  it("Shuffle Then sort", function (done) {
+    this.timeout(5000);
     let roll = 0;
-    let counts: { [k: number]: number } = {};
+    let counts: { [k: string]: number } = {};
 
-    for (let i = 1; i <= 5; i++) {
+    for (let i = 0; i < myArr.length; i++) {
       counts[i] = 0;
     }
 
     while (roll < 1000000) {
```

Comment:
Perhaps write a comment about how you actually test this. I'm not really following your test here.